### PR TITLE
[ANDROAPP-5355] Update rule engine to 2.1.7

### PR DESCRIPTION
**Pull request by andresmr**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-5355)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: form/src/main/java/org/dhis2/form/bindings/RuleExtensions.kt [hunk 1]
```diff
@@ -283,54 +286,103 @@ fun ProgramRuleVariable.toRuleVariable(
...
+        return emptyList()
+    }
+
+    if (dataElementUid != null) {
+        return dataElementRepository.uid(dataElementUid).blockingGet()?.optionSet()?.uid()
+            ?.let { optionSetUid ->
+                optionRepository.byOptionSetUid().eq(optionSetUid).blockingGet()
+            }?.map { option -> Option(option.name(), option.code()) } ?: emptyList()
+    } else if (trackedEntityAttributeUid != null) {
+        attributeRepository.uid(trackedEntityAttributeUid).blockingGet()?.optionSet()?.uid()
```
Comment (vgarciabnz) [inline form/src/main/java/org/dhis2/form/bindings/RuleExtensions.kt:377]: It seems there is a missing "return" here

File: stock-usecase/src/main/java/org/dhis2/android/rtsm/utils/RuleExtensions.kt [hunk 2]
```diff
@@ -135,56 +140,113 @@ fun ProgramRuleVariable.toRuleVariable(
...
+        return emptyList()
+    }
+
+    if (dataElementUid != null) {
+        return dataElementRepository.uid(dataElementUid).blockingGet()?.optionSet()?.uid()
+            ?.let { optionSetUid ->
+                optionRepository.byOptionSetUid().eq(optionSetUid).blockingGet()
+            }?.map { option -> Option(option.name(), option.code()) } ?: emptyList()
+    } else if (trackedEntityAttributeUid != null) {
+        attributeRepository.uid(trackedEntityAttributeUid).blockingGet()?.optionSet()?.uid()
```
Comment (vgarciabnz) [inline stock-usecase/src/main/java/org/dhis2/android/rtsm/utils/RuleExtensions.kt:241]: missing return

Comment (andresmr) [reply stock-usecase/src/main/java/org/dhis2/android/rtsm/utils/RuleExtensions.kt:241]: good catch!

File: form/src/main/java/org/dhis2/form/bindings/RuleExtensions.kt (hunk 1 above)
Comment (andresmr) [reply form/src/main/java/org/dhis2/form/bindings/RuleExtensions.kt:377]: fixed!

**Review by mmmateos — APPROVED** [review]

**Review by ferdyrod — APPROVED** [review]

**Pull request merged**